src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -570,11 +563,34 @@ private IAccount GetAccountByUserFlow(IEnumerable<IAccount> accounts, string use
             foreach (var account in accounts)
             {
                 string accountIdentifier = account.HomeAccountId.ObjectId.Split('.')[0];
-                if (accountIdentifier.EndsWith(userFlow.ToLower()))
+                if (accountIdentifier.EndsWith(userFlow.ToLower(CultureInfo.CurrentCulture), StringComparison.InvariantCulture))
                     return account;
             }
 
             return null;
         }
+
+        internal /*for test only*/ string CreateRedirectUri()
+        {
+            if (Uri.TryCreate(_microsoftIdentityOptions.RedirectUri, UriKind.Absolute, out Uri uri))
+            {
+                if (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps)
+                {
+                    return _microsoftIdentityOptions.RedirectUri;
+                }
+                else
+                {
+                    _logger.LogInformation("MicrosoftIdentityOptions RedirectUri value must have a Uri Scheme " +
```

Comment:
Wouldn't you want to throw an exception here?

Reply:
no because we can still figure out the redirect URI based on the httpContext. But logging the information. I guess this is more of question for JM>

---
In reply to: [420320742](https://github.com/AzureAD/microsoft-identity-web/pull/144#discussion_r420320742) [](ancestors = 420320742)